### ISSUE TITLE
DB import speed: Mysql 8.0 4x slower vs MariaDB 10.4

### ISSUE THREAD

**Issue by hkirsman** [ddev/ddev#6244]
### Preliminary checklist

- [X] I am using the latest stable version of DDEV (see [upgrade guide](https://ddev.readthedocs.io/en/stable/users/install/ddev-upgrade/))
- [X] I have searched [existing issues](https://github.com/ddev/ddev/issues)
- [X] I have checked the [troubleshooting guide](https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/)
- [X] I have run `ddev debug test` to include output below

### Output of `ddev debug test`

<details><summary>Expand `ddev debug test` diagnostic information</summary>

```


======== Please make sure that you have already looked at troubleshooting guide ========
Troubleshooting guide: https://ddev.readthedocs.io/en/stable/users/usage/troubleshooting/ 
Simple things to check:
* ddev poweroff
* Restart Docker Provider
* Reboot computer
* Temporarily disable VPN and firewall
Press any key to continue:


======== Output file will be in /tmp/ddev-debug-test.txt ========


======== Existing project config ========
ddev installation alternate locations:
/usr/bin/ddev
/bin/ddev

These config files were loaded for project example.com: [/home/hannes/Sites/example/.ddev/config.yaml /home/hannes/Sites/example/.ddev/config.wunderio.yaml]
name: example.com
type: drupal10
docroot: web
php_version: 8.1
webserver_type: nginx-fpm
webimage: ddev/ddev-webserver:v1.23.1
database: {mysql 8.0}
hooks: map[post-import-db:[map[exec:/var/www/html/.ddev/wunderio/core/_run-scripts.sh hooks-db-post-import.sh]] post-start:[map[exec-host:.ddev/wunderio/core/_run-scripts.sh hooks-host-post-start-2.sh] map[exec:[ ! -d "/var/www/html/vendor" ] && /var/www/html/.ddev/wunderio/core/_run-scripts.sh hooks-web-post-start-once.sh || true] map[exec-host:.ddev/wunderio/core/_run-scripts.sh hooks-host-post-start.sh] map[exec:/var/www/html/.ddev/wunderio/core/_run-scripts.sh hooks-web-post-start.sh] map[exec:UPDATE_SCRIPT="/var/www/html/vendor/wunderio/ddev-drupal/scripts/update_check.sh"; [ -f "$UPDATE_SCRIPT" ] && $UPDATE_SCRIPT || true]]]
project_tld: ddev.site
use_dns_when_possible: true
composer_version: 2.7.2
disable_settings_management: true
nodejs_version: 18
default_container_timeout: 180


======== existing project customizations ========
.ddev/docker-compose.mounts.yaml
.ddev/php/php-example.ini.example
.ddev/php/php.ini
.ddev/nginx_full/nginx-site.conf


======== installed DDEV add-ons ========
No registered add-ons were found. Add-ons installed before DDEV v1.22.0 will not be listed.
Update them with `ddev get` so they'll be shown.


======== mutagen situation ========
looking for #ddev-generated in mutagen.yml in project /home/hannes/Sites
.ddev/mutagen/mutagen.yml not found


======== Creating dummy project named tryddevproject-28389 in ../tryddevproject-28389 ========
Creating a new DDEV project config in the current directory (/home/hannes/Sites/tryddevproject-28389)
Once completed, your configuration will be written to /home/hannes/Sites/tryddevproject-28389/.ddev/config.yaml

Configuring a 'php' project with docroot 'web' at /home/hannes/Sites/tryddevproject-28389/web
Configuration complete. You may now run 'ddev start'.


======== OS Information ========
Linux hannes-ThinkPad-X1-Carbon-Gen-9 6.5.0-35-generic #35~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue May  7 09:00:52 UTC 2 x86_64 x86_64 x86_64 GNU/Linux


======== User information ========
uid=1000(hannes) gid=1000(hannes) groups=1000(hannes),4(adm),24(cdrom),27(sudo),30(dip),46(plugdev),120(lpadmin),132(lxd),133(sambashare),996(microk8s),998(docker)


======== DDEV version ========
 ITEM             VALUE                                    
 DDEV version     v1.23.1                                  
 architecture     amd64                                    
 cgo_enabled      0                                        
 db               ddev/ddev-dbserver-mariadb-10.11:v1.23.1 
 ddev-ssh-agent   ddev/ddev-ssh-agent:v1.23.1              
 docker           26.1.3                                   
 docker-api       1.45                                     
 docker-compose   v2.27.0                                  
 docker-platform  linux-docker                             
 mutagen          0.17.2                                   
 os               linux                                    
 router           ddev/ddev-traefik-router:v1.23.1         
 web              ddev/ddev-webserver:v1.23.1              



======== proxy settings ========

 HTTP_PROXY=''
 HTTPS_PROXY=''
 http_proxy=''
 NO_PROXY=''
 


======== DDEV global info ========
developer-mode=false
disable-http2=false
fail-on-hook-fail=false
instrumentation-opt-in=true
instrumentation-queue-size=0
instrumentation-reporting-interval=0s
instrumentation-user=
internet-detection-timeout-ms=3000
last-started-version=v1.23.1
letsencrypt-email=
mailpit-http-port=8025
mailpit-https-port=8026
mkcert-caroot=/home/hannes/.local/share/mkcert
no-bind-mounts=false
omit-containers=[]
performance-mode=none
project-tld=ddev.site
required-docker-compose-version=v2.27.0
router=traefik
router-bind-all-interfaces=false
router-http-port=80
router-https-port=443
simple-formatting=false
table-style=default
traefik-monitor-port=10999
use-docker-compose-from-path=false
use-hardened-images=false
use-letsencrypt=false
wsl2-no-windows-hosts-mgt=false
xdebug-ide-location=


======== DOCKER provider info ========
docker client location: -rwxr-xr-x 1 root root 39732136 mai   16 11:33 /usr/bin/docker

docker client alternate locations:
/usr/bin/docker
/bin/docker

Docker provider: linux-docker
docker version: 
Client: Docker Engine - Community
 Version:           26.1.3
 API version:       1.45
 Go version:        go1.21.10
 Git commit:        b72abbb
 Built:             Thu May 16 08:33:49 2024
 OS/Arch:           linux/amd64
 Context:           default

Server: Docker Engine - Community
 Engine:
  Version:          26.1.3
  API version:      1.45 (minimum version 1.24)
  Go version:       go1.21.10
  Git commit:       8e96db1
  Built:            Thu May 16 08:33:49 2024
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.32
  GitCommit:        8b3b7ca2e5ce38e8f31a34f35b2b68ceb8470d89
 runc:
  Version:          1.1.12
  GitCommit:        v1.1.12-0-g51d5e94
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0

DOCKER_DEFAULT_PLATFORM=notset
Docker platform: linux-docker
Using Docker context: default (unix:///var/run/docker.sock)
docker-compose: v2.27.0
Using DOCKER_HOST=unix:///var/run/docker.sock
Docker version: 26.1.3
Able to run simple container that mounts a volume.
Able to use internet inside container.
Docker disk space:Filesystem                Size      Used Available Use% Mounted on
overlay                 469.8G    353.0G     92.8G  79% /


======== Existing docker containers ========
CONTAINER ID   IMAGE                                        COMMAND                  CREATED        STATUS                     PORTS     NAMES
562b8a82216e   bitnami/nginx:1.25.3-debian-11-r1            "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_appserver_nginx_1
2ebc3125a7a5   mailhog/mailhog:v1.0.1                       "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_mailhog_1
ff6756b722e2   phpmyadmin/phpmyadmin:5.1.1                  "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_phpmyadmin_1
6b358ce81c39   dehy/adminer                                 "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_adminer_1
bfbde9e56e48   drupalci/webdriver-chromedriver:production   "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_chrome_1
7b7a7e9ce5bb   bitnami/memcached:1.6.1-debian-10-r4         "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_memcached_1
41e26d64fa12   devwithlando/php:8.1-fpm-4                   "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_appserver_1
c41a45e8a74a   bitnami/mysql:8.0.19-debian-10-r57           "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_database_1
67cdf3c7b7be   backstopjs/backstopjs:6.2.2                  "/lando-entrypoint.s…"   33 hours ago   Exited (137) 8 hours ago             examplecom_node_1
7760de656b79   bitnami/mysql:5.7.29-debian-10-r51           "/lando-entrypoint.s…"   3 days ago     Exited (137) 2 days ago              codequality_database_1
3c28155b3896   devwithlando/php:8.1-apache-4                "/lando-entrypoint.s…"   3 days ago     Exited (137) 2 days ago              codequality_appserver_1
7d0f982bada4   devwithlando/php:8.2-fpm-4                   "/lando-entrypoint.s…"   8 days ago     Exited (137) 8 days ago              downloadexamplecom_appserver_1
f80ab4746a0c   ppodgorsek/robot-framework                   "run-tests-in-virtua…"   9 days ago     Exited (1) 9 days ago                agitated_bohr
f3450615fdf3   ppodgorsek/robot-framework                   "run-tests-in-virtua…"   9 days ago     Exited (1) 9 days ago                crazy_bell
144b7552b0d4   ppodgorsek/robot-framework                   "run-tests-in-virtua…"   9 days ago     Exited (252) 9 days ago              optimistic_hopper
65b16674d1ec   ppodgorsek/robot-framework                   "run-tests-in-virtua…"   9 days ago     Exited (252) 9 days ago              xenodochial_galileo
2d6bc4846c00   ppodgorsek/robot-framework                   "run-tests-in-virtua…"   9 days ago     Exited (252) 9 days ago              adoring_wozniak
95832e099d33   robot-test                                   "./entrypoint.sh tes…"   9 days ago     Exited (1) 9 days ago                blissful_shockley
931c9358b2f9   fc3ed48e6e1d                                 "./entrypoint.sh tes…"   9 days ago     Created                              eloquent_goldstine
9714c34dff4f   traefik:2.2.0                                "/lando-entrypoint.s…"   10 days ago    Exited (137) 8 hours ago             landoproxyhyperion5000gandalfedition_proxy_1
ab74b3b4e128   bitnami/mysql:5.7.29-debian-10-r51           "/lando-entrypoint.s…"   11 days ago    Exited (137) 11 days ago             drupal10_database_1
a9e8dd1273b7   bitnami/nginx:1.18.0-debian-10-r363          "/lando-entrypoint.s…"   11 days ago    Exited (137) 11 days ago             drupal10_appserver_nginx_1
4f32c6aad7e7   devwithlando/php:8.2-fpm-4                   "/lando-entrypoint.s…"   11 days ago    Exited (137) 11 days ago             drupal10_appserver_1
2e967b0a1c08   devwithlando/php:8.1-fpm-4                   "/lando-entrypoint.s…"   11 days ago    Exited (137) 8 days ago              developerexamplecom_appserver_1
4a5977a6cf3e   bitnami/mysql:8.0.19-debian-10-r57           "/lando-entrypoint.s…"   11 days ago    Exited (137) 11 days ago             developerexamplecom_database_1
89d70feeec5a   node:16                                      "/lando-entrypoint.s…"   11 days ago    Exited (137) 11 days ago             developerexamplecom_node_1
e82faa70338d   localstack/localstack:latest                 "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_mock-aws_1
37e3b2a07afd   mailhog/mailhog:v1.0.1                       "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_mailhog_1
357654012d9b   bitnami/nginx:1.25.3-debian-11-r1            "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_appserver_nginx_1
cdf637ef5777   cars10/elasticvue:1.0.7                      "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_elasticvue_1
8bab3db19c29   bitnami/mysql:8.0.19-debian-10-r57           "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_database_1
a59b61cee7a8   opensearchproject/opensearch:2.11.1          "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_elasticsearch_1
3b897d80c15c   devwithlando/php:8.2-fpm-4                   "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_appserver_1
0fe15c00cd74   node:18                                      "/lando-entrypoint.s…"   2 weeks ago    Exited (137) 2 weeks ago             examplecom_node_1


======== mkcert information ========
/usr/bin/mkcert
/bin/mkcert
/home/hannes/.local/share/mkcert
total 8
-r-------- 1 hannes hannes 2484 dets   7 10:42 rootCA-key.pem
-rw-r--r-- 1 hannes hannes 1805 dets   7 10:42 rootCA.pem


======== ping attempt on ddev.site ========
PING dkdkd.ddev.site (127.0.0.1) 56(84) bytes of data.
64 bytes from localhost (127.0.0.1): icmp_seq=1 ttl=64 time=0.016 ms

--- dkdkd.ddev.site ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.016/0.016/0.016/0.000 ms


======== ddev debug refresh ========
Rebuilding project images...
Rebuilding web image with `/home/hannes/.ddev/bin/docker-compose -f /home/hannes/Sites/tryddevproject-28389/.ddev/.ddev-docker-compose-full.yaml build web --no-cache`
#0 building with "default" instance using docker driver

#1 [web internal] load build definition from Dockerfile
#1 transferring dockerfile: 1.23kB done
#1 DONE 0.0s

#2 [web internal] load metadata for docker.io/ddev/ddev-webserver:v1.23.1
#2 DONE 0.0s

#3 [web internal] load .dockerignore
#3 transferring context: 2B done
#3 DONE 0.0s

#4 [web 1/7] FROM docker.io/ddev/ddev-webserver:v1.23.1
#4 CACHED

#5 [web 2/7] RUN (groupadd --gid 1000 "hannes" || groupadd "hannes" || true) && (useradd  -l -m -s "/bin/bash" --gid "hannes" --comment '' --uid 1000 "hannes" || useradd  -l -m -s "/bin/bash" --gid "hannes" --comment '' "hannes" || useradd  -l -m -s "/bin/bash" --gid "1000" --comment '' "hannes" || useradd -l -m -s "/bin/bash" --comment '' hannes )
#5 DONE 0.4s

#6 [web 3/7] RUN export XDEBUG_MODE=off; composer self-update --stable || composer self-update --stable || true; composer self-update --2 || composer self-update --2 || true
#6 0.544 Storing "stable" as default update channel for the next self-update run.
#6 0.764 You are already using the latest available Composer version 2.7.6 (stable channel).
#6 1.068 You are already using the latest available Composer version 2.7.6 (2.x channel).
#6 DONE 1.1s

#7 [web 4/7] RUN mkdir -p /home/hannes && chown hannes /home/hannes && chmod 600 /home/hannes/.pgpass
#7 DONE 0.4s

#8 [web 5/7] RUN apt update
#8 0.350 
#8 0.350 WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
#8 0.350 
#8 0.465 Get:1 http://deb.debian.org/debian bookworm InRelease [151 kB]
#8 0.501 Get:2 http://deb.debian.org/debian bookworm-updates InRelease [55.4 kB]
#8 0.520 Get:3 http://deb.debian.org/debian-security bookworm-security InRelease [48.0 kB]
#8 0.556 Get:4 http://deb.debian.org/debian bookworm/main amd64 Packages [8786 kB]
#8 0.648 Get:5 https://packages.sury.org/php bookworm InRelease [7542 B]
#8 0.677 Get:6 https://deb.nodesource.com/node_20.x nodistro InRelease [12.1 kB]
#8 0.782 Get:7 https://packages.sury.org/php bookworm/main amd64 Packages [229 kB]
#8 0.819 Get:8 http://nginx.org/packages/debian bookworm InRelease [2860 B]
#8 0.841 Get:9 https://dl.cloudsmith.io/public/platformsh/cli/deb/debian bookworm InRelease [3675 B]
#8 0.848 Get:10 https://deb.nodesource.com/node_20.x nodistro/main amd64 Packages [7455 B]
#8 0.917 Get:11 http://nginx.org/packages/debian bookworm/nginx amd64 Packages [7975 B]
#8 1.010 Get:12 https://dl.cloudsmith.io/public/platformsh/upsun-cli/deb/debian bookworm InRelease [3675 B]
#8 1.139 Get:13 http://packages.blackfire.io/debian any InRelease [28.0 kB]
#8 1.219 Get:14 https://dl.cloudsmith.io/public/symfony/stable/deb/debian bookworm InRelease [4422 B]
#8 1.442 Get:15 https://dl.cloudsmith.io/public/platformsh/cli/deb/debian bookworm/main amd64 Packages [3310 B]
#8 1.552 Get:16 https://dl.cloudsmith.io/public/platformsh/upsun-cli/deb/debian bookworm/main amd64 Packages [2193 B]
#8 1.672 Get:17 https://dl.cloudsmith.io/public/symfony/stable/deb/debian bookworm/main amd64 Packages [6917 B]
#8 2.100 Get:18 http://deb.debian.org/debian bookworm-updates/main amd64 Packages [13.8 kB]
#8 2.103 Get:19 http://deb.debian.org/debian-security bookworm-security/main amd64 Packages [157 kB]
#8 2.411 Get:20 http://packages.blackfire.io/debian any/main amd64 Packages [26.2 kB]
#8 2.740 Fetched 9557 kB in 2s (4048 kB/s)
#8 2.740 Reading package lists...
#8 3.098 Building dependency tree...
#8 3.195 Reading state information...
#8 3.210 7 packages can be upgraded. Run 'apt list --upgradable' to see them.
#8 DONE 3.3s

#9 [web 6/7] RUN curl -I https://www.google.com
#9 0.316   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
#9 0.316                                  Dload  Upload   Total   Spent    Left  Speed
#9 0.316 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0HTTP/2 200 
#9 0.405 content-type: text/html; charset=ISO-8859-1
#9 0.405 content-security-policy-report-only: object-src 'none';base-uri 'self';script-src 'nonce-4JX1N7Fq41gXlyRQj8ykJg' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
#9 0.405 p3p: CP="This is not a P3P policy! See g.co/p3phelp for more info."
#9 0.405 date: Fri, 24 May 2024 19:00:33 GMT
#9 0.405 server: gws
#9 0.405 x-xss-protection: 0
#9 0.405 x-frame-options: SAMEORIGIN
#9 0.405 expires: Fri, 24 May 2024 19:00:33 GMT
#9 0.405 cache-control: private
#9 0.405 set-cookie: AEC=AQTF6HzEQ7EgyE-41FSvo9q8QEI8cL0V-TQUx5twYxcA8u2caS6plMCzE9M; expires=Wed, 20-Nov-2024 19:00:33 GMT; path=/; domain=.google.com; Secure; HttpOnly; SameSite=lax
#9 0.405 set-cookie: __Secure-ENID=19.SE=AxEHaA7J0520NnHefMimZGqrGnaY213cdYCw-VGAgMBpYHxgmdY0FNhNY7r4wJ6lr7k75VnBSfapRodEB3qHS-lhah9hUdZXKUkHvELAfPgCXwqPcS0W6dXcAvUlcWQQwOvBCTcj6pxhQbkZcAsRakr2wfdcrYCmv9DdnfY_--ru5JN5gpWHew; expires=Tue, 24-Jun-2025 11:18:51 GMT; path=/; domain=.google.com; Secure; HttpOnly; SameSite=lax
#9 0.405 alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000
#9 0.405 
#9 0.405 
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
#9 DONE 0.4s

#10 [web 7/7] RUN chmod 777 /run/php
#10 DONE 0.4s

#11 [web] exporting to image
#11 exporting layers
#11 exporting layers 0.2s done
#11 writing image sha256:a182b079a3b537cd6833661242e7611d99e51fcdc7d41d30350aeb6c6cb0d134 done
#11 naming to docker.io/ddev/ddev-webserver:v1.23.1-tryddevproject-28389-built 0.0s done
#11 DONE 0.3s
Refreshed Docker cache for project tryddevproject-28389 in 7s
Network ddev_default created
 Container ddev-ssh-agent  Created
 Container ddev-ssh-agent  Started
ssh-agent container is running: If you want to add authentication to the ssh-agent container, run 'ddev auth ssh' to enable your keys.
v1.23.1: Pulling from ddev/ddev-dbserver-mariadb-10.11
a8b1c5f80c2d: Pulling fs layer
d089a232b283: Pulling fs layer
e6a7853251a9: Pulling fs layer
a02c50173b46: Pulling fs layer
60cf01d97b1c: Pulling fs layer
8712b8a31288: Pulling fs layer
e2713dbe3052: Pulling fs layer
f0ffb22994e6: Pulling fs layer
4f4fb700ef54: Pulling fs layer
faa786aca2f0: Pulling fs layer
673fa2490007: Pulling fs layer
58fdfe1cd816: Pulling fs layer
fa157a347f2a: Pulling fs layer
106f935e70a6: Pulling fs layer
a02c50173b46: Waiting
cfb652f93b52: Pulling fs layer
5851332ba810: Pulling fs layer
60cf01d97b1c: Waiting
5399635ca190: Pulling fs layer
8712b8a31288: Waiting
4f4fb700ef54: Waiting
bade098f2205: Pulling fs layer
f601a7f30211: Pulling fs layer
e2713dbe3052: Waiting
94f7e7becbb2: Pulling fs layer
faa786aca2f0: Waiting
f0ffb22994e6: Waiting
a84f9b9b9383: Pulling fs layer
673fa2490007: Waiting
db780a914fb6: Pulling fs layer
ab859101a885: Pulling fs layer
343b39bf175e: Pulling fs layer
5851332ba810: Waiting
58fdfe1cd816: Waiting
5399635ca190: Waiting
f601a7f30211: Waiting
94f7e7becbb2: Waiting
ab859101a885: Waiting
db780a914fb6: Waiting
a84f9b9b9383: Waiting
cfb652f93b52: Waiting
106f935e70a6: Waiting
fa157a347f2a: Waiting
d089a232b283: Verifying Checksum
d089a232b283: Download complete
a02c50173b46: Download complete
e6a7853251a9: Verifying Checksum
e6a7853251a9: Download complete
60cf01d97b1c: Verifying Checksum
60cf01d97b1c: Download complete
e2713dbe3052: Verifying Checksum
e2713dbe3052: Download complete
f0ffb22994e6: Download complete
4f4fb700ef54: Download complete
faa786aca2f0: Verifying Checksum
faa786aca2f0: Download complete
a8b1c5f80c2d: Download complete
58fdfe1cd816: Download complete
a8b1c5f80c2d: Pull complete
d089a232b283: Pull complete
e6a7853251a9: Pull complete
a02c50173b46: Pull complete
60cf01d97b1c: Pull complete
fa157a347f2a: Verifying Checksum
fa157a347f2a: Download complete
106f935e70a6: Verifying Checksum
106f935e70a6: Download complete
cfb652f93b52: Verifying Checksum
cfb652f93b52: Download complete
5851332ba810: Verifying Checksum
5851332ba810: Download complete
673fa2490007: Verifying Checksum
673fa2490007: Download complete
bade098f2205: Download complete
f601a7f30211: Verifying Checksum
f601a7f30211: Download complete
5399635ca190: Verifying Checksum
5399635ca190: Download complete
94f7e7becbb2: Verifying Checksum
94f7e7becbb2: Download complete
db780a914fb6: Verifying Checksum
db780a914fb6: Download complete
a84f9b9b9383: Verifying Checksum
a84f9b9b9383: Download complete
ab859101a885: Download complete
343b39bf175e: Verifying Checksum
343b39bf175e: Download complete
8712b8a31288: Verifying Checksum
8712b8a31288: Download complete
8712b8a31288: Pull complete
e2713dbe3052: Pull complete
f0ffb22994e6: Pull complete
4f4fb700ef54: Pull complete
faa786aca2f0: Pull complete
673fa2490007: Pull complete
58fdfe1cd816: Pull complete
fa157a347f2a: Pull complete
106f935e70a6: Pull complete
cfb652f93b52: Pull complete
5851332ba810: Pull complete
5399635ca190: Pull complete
bade098f2205: Pull complete
f601a7f30211: Pull complete
94f7e7becbb2: Pull complete
a84f9b9b9383: Pull complete
db780a914fb6: Pull complete
ab859101a885: Pull complete
343b39bf175e: Pull complete
Digest: sha256:cd01a6e8e211e5d09d9a9d2fa57a6de5d4d27a4a08bc7c2752a0dc6376884f3a
Status: Downloaded newer image for ddev/ddev-dbserver-mariadb-10.11:v1.23.1
docker.io/ddev/ddev-dbserver-mariadb-10.11:v1.23.1
Building project images...
..Project images built in 2s.
 Network ddev-tryddevproject-28389_default  Created
 Container ddev-tryddevproject-28389-web  Created
 Container ddev-tryddevproject-28389-db  Created
 Container ddev-tryddevproject-28389-db  Started
 Container ddev-tryddevproject-28389-web  Started
Waiting for web/db containers to become ready: [web db]
Starting ddev-router if necessary...
 Container ddev-router  Created
 Container ddev-router  Started
Waiting for additional project containers to become ready...
All project containers are now ready.


======== Project startup ========
Starting tryddevproject-28389...
2024-05-24T22:01:34.375 Pulled image for ddev/ddev-webserver:v1.23.1
2024-05-24T22:01:34.375 Pulled image for busybox:stable
2024-05-24T22:01:34.376 Pulled image for ddev/ddev-ssh-agent:v1.23.1
2024-05-24T22:01:34.376 Pulled image for ddev/ddev-traefik-router:v1.23.1
2024-05-24T22:01:35.501 chowning /mnt/ddev-global-cache and /var/lib/mysql to 1000
2024-05-24T22:01:36.11 done chowning /mnt/ddev-global-cache and /var/lib/mysql to 1000
2024-05-24T22:01:36.115 host.docker.internal='' runtime.GOOS==linux and docker 20.10+
2024-05-24T22:01:36.157 Pulled image for ddev/ddev-dbserver-mariadb-10.11:v1.23.1
2024-05-24T22:01:36.157 Pulled image for ddev/ddev-webserver:v1.23.1
2024-05-24T22:01:36.158 Pulled image for ddev/ddev-ssh-agent:v1.23.1
2024-05-24T22:01:36.158 Pulled image for ddev/ddev-traefik-router:v1.23.1
Building project images...
2024-05-24T22:01:36.17 Executing docker-compose -f /home/hannes/Sites/tryddevproject-28389/.ddev/.ddev-docker-compose-full.yaml build --progress=plain
.Project images built in 1s.
2024-05-24T22:01:37.172 Executing docker-compose -f /home/hannes/Sites/tryddevproject-28389/.ddev/.ddev-docker-compose-full.yaml up -d
 Container ddev-tryddevproject-28389-web  Recreate
 Container ddev-tryddevproject-28389-db  Recreate
 Container ddev-tryddevproject-28389-db  Recreated
 Container ddev-tryddevproject-28389-web  Recreated
 Container ddev-tryddevproject-28389-web  Started
 Container ddev-tryddevproject-28389-db  Started
2024-05-24T22:01:48.919 VIRTUAL_HOST=tryddevproject-28389.ddev.site for web
2024-05-24T22:01:48.919 HTTP_EXPOSE=80:80,8025:8025 for web
2024-05-24T22:01:48.919 HTTPS_EXPOSE=443:80,8026:8025 for web
2024-05-24T22:01:49.446 Copied /home/hannes/Sites/tryddevproject-28389/.ddev/traefik:CopyIntoVolume_sstfhmrhqdwx into /mnt/v/traefik in 63.089021ms
2024-05-24T22:01:49.475 Exec chown -R 1000 /mnt/v/traefik stdout=, stderr=, err=<nil>
2024-05-24T22:01:49.699 Copied Traefik certs in /home/hannes/Sites/tryddevproject-28389/.ddev/traefik/certs to ddev-global-cache/traefik
2024-05-24T22:01:49.699 Running /start.sh in ddev-webserver
Waiting for web/db containers to become ready: [web db]
2024-05-24T22:01:52.763 Copied /home/hannes/.ddev/commands:CopyIntoVolume_lykmyfaocnzi into /mnt/v/global-commands in 73.191772ms
2024-05-24T22:01:52.787 Exec chown -R 1000 /mnt/v/global-commands stdout=, stderr=, err=<nil>
2024-05-24T22:01:53.093 Testing to see if /mnt/ddev_config is properly mounted
Starting ddev-router if necessary...
2024-05-24T22:01:53.653 Copied /home/hannes/.ddev/traefik:CopyIntoVolume_zulekudoqnkc into /mnt/v/traefik in 54.268116ms
2024-05-24T22:01:53.677 Exec chown -R 1000 /mnt/v/traefik stdout=, stderr=, err=<nil>
2024-05-24T22:01:53.928 Copied global Traefik config in /home/hannes/.ddev/traefik/certs to ddev-global-cache/traefik
 Container ddev-router  Running
2024-05-24T22:01:53.961 Waiting for ddev-router to become ready. docker inspect --format "{{json .State.Health }}" ddev-router
2024-05-24T22:01:54.465 ddev-router is ready
Waiting for additional project containers to become ready...
All project containers are now ready.
Successfully started tryddevproject-28389
Project can be reached at https://tryddevproject-28389.ddev.site https://127.0.0.1:32888


======== Curl of site from inside container ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 200 OK
Server: nginx
Date: Fri, 24 May 2024 19:01:56 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding



======== curl -I of http://127.0.0.1:32889 (web container http docker bind port) from outside ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 200 OK
Server: nginx
Date: Fri, 24 May 2024 19:01:56 GMT
Content-Type: text/html; charset=UTF-8
Connection: keep-alive
Vary: Accept-Encoding



======== curl -I of http://tryddevproject-28389.ddev.site (router http URL) from outside ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 200 OK
Content-Type: text/html; charset=UTF-8
Date: Fri, 24 May 2024 19:01:56 GMT
Server: nginx
Vary: Accept-Encoding



======== Full curl of http://tryddevproject-28389.ddev.site (router http URL) from outside ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   198    0   198    0     0  14213      0 --:--:-- --:--:-- --:--:-- 15230
Success accessing database... db via TCP/IP<br />
ddev is working.<br />
The output file for Discord or issue queue is in
<b>/tmp/ddev-debug-test.txt</b><br />
file:///tmp/ddev-debug-test.txt<br />


======== Full curl of https://tryddevproject-28389.ddev.site (router https URL) from outside ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100   198    0   198    0     0   4185      0 --:--:-- --:--:-- --:--:--  4212
Success accessing database... db via TCP/IP<br />
ddev is working.<br />
The output file for Discord or issue queue is in
<b>/tmp/ddev-debug-test.txt</b><br />
file:///tmp/ddev-debug-test.txt<br />


======== Curl google.com to check internet access and VPN ========
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed

  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/2 200 
content-type: text/html; charset=ISO-8859-1
content-security-policy-report-only: object-src 'none';base-uri 'self';script-src 'nonce-BamjlQG87rSGP--gHjt_ig' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
p3p: CP="This is not a P3P policy! See g.co/p3phelp for more info."
date: Fri, 24 May 2024 19:01:56 GMT
server: gws
x-xss-protection: 0
x-frame-options: SAMEORIGIN
expires: Fri, 24 May 2024 19:01:56 GMT
cache-control: private
set-cookie: AEC=AQTF6HwpXD9KkickOLgmfaiNGPdrQRqHX3aohdz_Do_3Ao51i8x6XgH8reY; expires=Wed, 20-Nov-2024 19:01:56 GMT; path=/; domain=.google.com; Secure; HttpOnly; SameSite=lax
set-cookie: __Secure-ENID=19.SE=LFnvnhO9sRWVurppLNdftix64BHSZiLhalZIPIOPdy6yfgYAQZW-S439vHj131Sdi6tyDme06hu-0CFjPA8ApGkUxc5CdKKbsYmb1DnjtqrP3AKD3J6K_60Kjh1tpebTP_58F9PXkuoPo-pILXVb1M6ocFLjt4EKQdb5C9o6rCohix1b2zEdIA; expires=Tue, 24-Jun-2025 11:20:14 GMT; path=/; domain=.google.com; Secure; HttpOnly; SameSite=lax
alt-svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000



======== host.docker.internal status ========
PING host.docker.internal (172.17.0.1) 56(84) bytes of data.
64 bytes from host.docker.internal (172.17.0.1): icmp_seq=1 ttl=64 time=0.049 ms

--- host.docker.internal ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 0.049/0.049/0.049/0.000 ms


======== Project ownership on host ========
drwxrwxr-x 4 hannes hannes 4096 mai   24 22:00 ../tryddevproject-28389


======== Project ownership in container ========
drwxrwxr-x 4 hannes hannes 4096 May 24 19:00 //var/www/html


======== In-container filesystem ========
Filesystem     Type 1K-blocks      Used Available Use% Mounted on
/dev/nvme0n1p3 ext4 489161968 435907112  28333512  94% /var/www/html


======== Thanks for running the diagnostic! ========
Running ddev launch in 3 seconds
Running ddev launch
Waiting for ddev launch to complete before deleting project
Opening in existing browser session.


Cleanup: deleting test project tryddevproject-28389
 Container ddev-tryddevproject-28389-db  Stopped
```
</details>

### Expected Behavior

I expect MySQL 8 to import as fast as MariaDB or at least as close to it. Right now the difference is too big. I imported 3x times 14GB mysql file and the import speed with Maria DB was:
1. 18m 24s
2. 15m 53s
3. 15m 56s

### Actual Behavior

I didn't have time to do 3 tests atm, but the first test with MySQL 8 was:
1. 59m 16s

This is something I have felt that's wrong somewhere, but I wasn't sure what. We changed to MySQL 1-2 months ago and previously I was used to 10-15 minutes import time for the big SQL with MariaDB.

### Steps To Reproduce

1. Nothing special in my env:

```
name: example.com
type: drupal10
docroot: web
php_version: "8.1"
webserver_type: nginx-fpm
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: []
database:
    type: mysql
    version: "8.0"
use_dns_when_possible: true
composer_version: "2.7.2"
disable_settings_management: true
web_environment: []
nodejs_version: "18"
# Fix "Failed waiting for web/db containers to become ready" when restoring snapshot.
# @todo Still still getting 2 minute warning so let's deal with this later.
default_container_timeout: "180"


```

2. This is script that I tested with:

```
#!/bin/bash

# Testing import speed

# Test 1
ddev delete --omit-snapshot --yes && ddev start
time ddev import-db --file=./db.sql.gz

# Test 2
ddev delete --omit-snapshot --yes && ddev start
time ddev import-db --file=./db.sql.gz

# Test 3
ddev delete --omit-snapshot --yes && ddev start
time ddev import-db --file=./db.sql.gz
```

**Comment by rfay** [comment]
I think you should probably open an issue with MySQL rather than DDEV :)

It also makes a lot of difference what Docker provider you're using, as that is key here, as @douggreen has demonstrated many times.

Also, MariaDB 10.4 is nearly obsolete, you should be comparing with MariaDB 10.11 if anything. But MySQL 8.0 is more complex than older MariaDB and I would expect some difference.

You also seem to have a post-import-db hook. I would expect that to impact your tests.

And you have *lots* of customizations. Please do exploration like this on a plain vanilla project

```
.ddev/docker-compose.mounts.yaml
.ddev/php/php-example.ini.example
.ddev/php/php.ini
.ddev/nginx_full/nginx-site.conf
```

and you have `.ddev/config.wunderio.yaml`

You don't mention how much memory you have on your thinkpad, which can also be a factor.

Closing this for now, please continue your exploration, I'd love to hear more, and I invite you to do a guest blog post on ddev.com with your results.

**Comment by hkirsman** [comment]
I know in Lando they do some overrides for database to speed things up - at least for the writing part. One was this https://dev.mysql.com/doc/refman/8.0/en/innodb-parameters.html#sysvar_innodb_flush_log_at_trx_commit and that speeds up things like 2x times if you change it from default 1 to either 0 or 2. So you're saying you just get default database image and don't adjust settings for MariaDB for example? 

I did have db post hook, but I emptied the script. But I do understand why testing this in vanilla would be better.

The slowness can be seen already from the import status. With MySQL 8 it jumps to 2+ hours and stays there for long time:
![2024-05-24_14-08](https://github.com/ddev/ddev/assets/492375/15d41925-18d0-4591-8f01-c7fa08689304)

With MariaDB it's more optimistic:
![2024-05-24_14-16](https://github.com/ddev/ddev/assets/492375/7814d683-6cae-48aa-99c6-6f8073a848c8)

**Comment by rfay** [comment]
DDEV's configuration is at https://github.com/ddev/ddev/blob/master/containers/ddev-dbserver/files/etc/my.cnf

You can also see it by `ddev ssh -s db` and view /etc/.cnf (use `cat /etc/my.cnf` or `vi /etc/my.cnf`

You can override it, see https://ddev.readthedocs.io/en/stable/users/extend/customization-extendibility/#custom-mysqlmariadb-configuration-mycnf, and anything you learn will be welcome. 

Sometimes when you import a database dump like this, the indexes get built as it loads, which is quite costly, especially on certain types of database. That is why DDEV has snapshots, which are in binary form and can be loaded without index creation.

Note that the actual structure of the database you're loading may be related to the behavior you're seeing, and of course the priorities of one database vendor over the other. 

If you haven't already read https://ddev.readthedocs.io/en/stable/users/usage/database-management/ please do, thanks!

**Comment by hkirsman** [comment]
I did another benchmark on Vanilla DDEV and latest Maria DB. I also unpacked the db to only count importing time.

MariaDB
Test 1: 10m 27s
Test 2: 9m 54s
Test 3:  9m 49s

MySQL 8
Test 1: 60m 54s
Test 2: 59m 18s
Test 3: 59m 35s

So the difference is not 4x times now, but 6x.

I created setup with 2 branches in this repo: https://github.com/hkirsman/ddev-db-import-bench

I guess one reason I'm here because snapshot tool is also broken atm as it times out on large db's. I understand it's in roadmap for next minor version.

**Comment by douggreen** [comment]
Snapshot doesn’t actually timeout. You should tail the log withddev logs -fs dbAnd wait for it to complete.What’s broken is setting the timeout that would allow this to run in the shell and not return until it is complete.My imports take over an hour (with mariadb). I turn off indexing during the import and turn it back on afterwards. Snapshot saves me hours every day.DougSent from my iPhonePlease excuse brevity and typosOn May 25, 2024, at 3:32 PM, Hannes Kirsman ***@***.***> wrote:﻿
I did another benchmark on Vanilla DDEV and latest Maria DB. I also unpacked the db to only count importing time.
MariaDB
Test 1: 10m 27s
Test 2: 9m 54s
Test 3:  9m 49s
MySQL 8
Test 1: 60m 54s
Test 2: 59m 18s
Test 3: 59m 35s
So the difference is not 4x times now, but 6x.
I created setup with 2 branches in this repo: https://github.com/hkirsman/ddev-db-import-bench
I guess one reason I'm here because snapshot tool is also broken atm as it times out on large db's. I understand it's in roadmap for next minor version.
Tx for the info!

**Comment by rfay** [comment]
As Doug says, the problem is just that the timeout isn't working correctly, 
* https://github.com/ddev/ddev/issues/5133

If you `ddev logs -s db -f` you'll see that `ddev snapshot restore` continues on just fine.

I'm glad you're exploring this territory, and you'll probably want to look more broadly.

* Are there issues in mariadb or mysql bug trackers about the difference?
* Is this a problem just for your particular database, or does this happen to any database? (For example, if a flat set of rows in one table with no index, what happens?)
* Maybe Doug's technique "I turn off indexing during the import and turn it back on afterwards" will help you as well.

**Comment by douggreen** [comment]
I have these hooks on my sites with large databases. tbh, I added these
some time ago, and have not re-evaluated if they are helpful.

hooks:
  pre-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE DISABLE INNODB REDO_LOG;"
  post-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE ENABLE INNODB REDO_LOG;"

**Comment by rfay** [comment]
I hope you can figure out the difference! It will be great. You can obviously use the lando mysql config in your .ddev/mysql/<configini> to test this. If you would add a link to the actual lando mysql config here, in your repo or a gist or whatever, or if it's short right in the issue or as an attached file.

**Comment by hkirsman** [comment]
In another site lot smaller db I got this result when trying Landos mysql conf (it's lot faster):

1. Default
ddev import-db --file=./down-prod-2024-05-16.sql.gz               
mysql: [Warning] Using a password on the command line interface can be insecure.
2.44GiB **0:14:01** [2.97MiB/s] [====================================================>] 100% 


2. dev with https://github.com/lando/drupal/blob/main/config/drupal10/mysql8.cnf

 ✗ ddev import-db --file=./down-prod-2024-05-16.sql.gz
mysql: [Warning] Using a password on the command line interface can be insecure.
2.44GiB **0:10:56** [3.81MiB/s] [====================================================>] 100% 

**But still with Lando it's like 5 minutes or less.**.. only difference I can think of is that Lando is using bitnami. Should I try to create custom service and try something there? Or should I try come up with dockerfile that directly spins up the containers and test there? What's the mysql 8 image DDEV is using?

**Comment by hkirsman** [comment]
I think I got it. Listed image in my system is this:
ddev/ddev-dbserver-mysql-8.0

Dockerhub https://hub.docker.com/r/ddev/ddev-dbserver-mysql-8.0

Github https://github.com/ddev/ddev/tree/master/containers/ddev-dbserver

**Comment by rfay** [comment]
Would love to have you try it with bitnami, it's pretty easy to do. Related:
* https://github.com/ddev/ddev/pull/5640

Bitnami only recently started providing arm64 images, and that made it so we might be able to use it as an upstream in the future.

Currently, we use for amd64 (your use) the official mysql image. For arm64 we have to build our own using Ubuntu base. 

My prediction would be that the difference will be the my.cnf/mysql.cnf, but your explorations are excellent.

**Comment by hkirsman** [comment]
So somewhere you check out that branch and then you have bitnami... but where in your system? How do you develop DDEV?

**Comment by rfay** [comment]
That PR didn't work out and never matured enough for you to try out. But you can use the bitnami/mysql image for testing (and compare it directly against the mysql:8.0 official image as well), see https://hub.docker.com/r/bitnami/mysql

**Comment by rfay** [comment]
For more about DDEV development, see https://ddev.readthedocs.io/en/stable/developers/

The image management stuff is in https://ddev.readthedocs.io/en/stable/developers/release-management/

**Comment by rfay** [comment]
I'm going to reopen this since you're doing such a good job exploring it. Thanks for the concerted effort!

**Comment by hkirsman** [comment]
Tx!

Got these results now:
13m 9s
12m 16s
11m 56s


The config is a bit of mess, put together from 3 different locations (mainly from Land), but it's good start:

```
#
# The MySQL database server configuration file for Lando
#
# LANDODRUPALMYSQL8CNF

[mysqld]


#from bitnami container
skip-name-resolve
explicit_defaults_for_timestamp
#basedir=/opt/bitnami/mysql
port=3306
#tmpdir=/opt/bitnami/mysql/tmp
#socket=/opt/bitnami/mysql/tmp/mysql.sock
#pid-file=/opt/bitnami/mysql/tmp/mysqld.pid
max_allowed_packet=16M
#bind-address=0.0.0.0
#log-error=/opt/bitnami/mysql/logs/mysqld.log
character-set-server=UTF8
collation-server=utf8_general_ci
#plugin_dir=/opt/bitnami/mysql/plugin



# from github


#
# * Basic Settings
#
# Data is stored in a volume on the db container /sql
default-storage-engine = innodb

#
# * Fine Tuning
#
key_buffer_size    = 384M
max_allowed_packet  = 32M
thread_stack    = 400K
thread_cache_size       = 8
# This replaces the startup script and checks MyISAM tables if needed
# the first time they are touched
#max_connections        = 100
#table_cache            = 64
#thread_concurrency     = 10
read_rnd_buffer_size = 8M
myisam_sort_buffer_size = 64M
table_open_cache = 512
sort_buffer_size = 2M
read_buffer_size = 2M

#
# * Logging and Replication
#
# Both location gets rotated by the cronjob.
# Be aware that this log type is a performance killer.
# As of 5.1 you can enable the log at runtime!
#general_log_file        = /src/.lando/log/mysql.log
#general_log             = 1
#
# Error logging goes to syslog due to /etc/mysql/conf.d/mysqld_safe_syslog.cnf.
#
# Here you can see queries with especially long duration
#log_slow_queries = /var/log/mysql/mysql-slow.log
#long_query_time = 2
#log-queries-not-using-indexes
#
# The following can be used as easy to replay backup logs or for replication.
# note: if you are setting up a replication slave, see README.Debian about
#       other settings you may need to change.
#server-id    = 1
#log_bin      = /src/.lando/log/mysql-bin.log
expire_logs_days = 10
max_binlog_size = 101M
#binlog_do_db   = include_database_name
#binlog_ignore_db = include_database_name
#
# * InnoDB
#
# InnoDB is enabled by default with a 10MB datafile in /var/lib/mysql/.
# Read the manual for more InnoDB related options. There are many!
#
# Uncomment the following if you are using InnoDB tables
#innodb_data_file_path = ibdata1:2000M;ibdata2:10M:autoextend
#innodb_log_group_home_dir = C:\mysql\data/
# You can set .._buffer_pool_size up to 50 - 80 %
# of RAM but beware of setting memory usage too high
#innodb_buffer_pool_size = 384M
#innodb_additional_mem_pool_size = 20M
# Set .._log_file_size to 25 % of buffer pool size
innodb_log_file_size = 100M
#innodb_log_buffer_size = 8M
innodb_flush_log_at_trx_commit = 0
#innodb_lock_wait_timeout = 50
innodb_buffer_pool_size = 384M
innodb_log_buffer_size  = 4M
innodb_file_per_table   = 1
innodb_open_files       = 256
innodb_io_capacity      = 512
innodb_flush_method     = O_DIRECT
innodb_thread_concurrency = 8
innodb_lock_wait_timeout = 127
#
# * Security Features
#
# Read the manual, too, if you want chroot!
# chroot = /var/lib/mysql/
#
# For generating SSL certificates I recommend the OpenSSL GUI "tinyca".
#
# ssl-ca=/etc/mysql/cacert.pem
# ssl-cert=/etc/mysql/server-cert.pem
# ssl-key=/etc/mysql/server-key.pem


# https://dba.stackexchange.com/questions/83125/mysql-any-way-to-import-a-huge-32-gb-sql-dump-faster

innodb_buffer_pool_size = 4G
innodb_log_buffer_size = 256M
innodb_log_file_size = 1G
innodb_write_io_threads = 16
innodb_flush_log_at_trx_commit = 0
innodb_doublewrite = 0

[mysqldump]
quick
quote-names
max_allowed_packet  = 32M

[mysql]
#no-auto-rehash # faster start of mysql but no tab completion

[isamchk]
key_buffer_size = 384M
sort_buffer_size = 256M
read_buffer = 2M
write_buffer = 2M

```

**Comment by rfay** [comment]
Looking forward to a .ddev/mysql/import.cnf that demonstrates the fix! And then a PR.

I don't understand `isamchk` here as ISAM is long obsolete and shouldn't be a part of anything... We're using Innodb...

**Comment by hkirsman** [comment]
I've done lot of testing and this is what I've come up with:
```
innodb_write_io_threads = 16
innodb_doublewrite = 0
innodb_log_file_size = 100MB
```

Originally the import takes 60 minutes. With the above config it takes 13.5 minutes.

Most impact gives innodb_doublewrite = 0. From 60 minutes to 15.3 minutes.

With innodb_log_file_size I tested with 1MB, 10MB, 20MB etc until 90MB because originally I had it at 1000 MB:
![image](https://github.com/ddev/ddev/assets/492375/1441e166-272c-4796-9ce3-034dc505d074)

At 70MB the import speed did not increase. According to MariaDB https://mariadb.com/kb/en/innodb-system-variables/#innodb_log_file_size that value is:
100663296 (96MB) (>= MariaDB 10.5), 50331648 (48MB) (<= MariaDB 10.4)

So that's why I've rounded it 100MB, but maybe 96MB would better? 

Checking my report, I think I need to test this again without setting innodb_write_io_threads at all.

**Comment by rfay** [comment]
Super interesting! Thanks for the good work. Could you please update the title and OP of the issue to reflect your current comparisons, and consider a PR when you're ready.

**Comment by douggreen** [comment]
I am starting a test with these settings. Isthe innodb log size is helpful if you disable logging during port.I don’t have a good history of my tests. I believe that switching to orb drastically slowed down my imports. My last one took almost 3 hours and I perceive that they used to be closer to an hour.I’ll let you know what it is with these settings…Sent from my iPhonePlease excuse brevity and typosOn Jun 16, 2024, at 8:44 AM, Randy Fay ***@***.***> wrote:﻿
Super interesting! Thanks for the good work. Could you please update the title and OP of the issue to reflect your current comparisons, and consider a PR when you're ready.

**Comment by douggreen** [comment]
This seems to have radically improved the actual time from start to finish, but I'm confused that the user/system time didn't change much. Is this real?

Before 112.55s user 12.05s system 1% cpu 2:44:06.85 total
After 108.91s user 10.44s system 1% cpu 1:59:04.68 total

**Comment by hkirsman** [comment]
So it's 2 hours and 44 minutes initially and then 2 hours, right? I would have hoped even more speed increase. I'm not sure if you did add all 3 settings or not?

The user is basically 2 minutes and system only around 10 seconds. There's not much to change to in my oppionion. Also I've asked my colleague and he said it's how long 'client' (mysql or ddev) took cpu cycles, but stops counting after the job is sent off to import. The thing to focus on is cpu.

**Comment by douggreen** [comment]
And on a second (supposedly identical) run, that 2:44 was now 2:15, so it was not consistent. I'll try with the log file size tonight after work.

**Comment by rfay** [comment]
Just a note: It's only the elapsed time that matters. User is user-level CPU, System is system-level CPU, but almost all of an import is waiting for I/O, so that's where the rest of the time is. DB imports aren't about CPU.

**Comment by hkirsman** [comment]
I always destroyed the instance before starting new test [douggreen](https://github.com/douggreen)

**Comment by douggreen** [comment]
I may have missed that step on the first test. I didn't on the second.

ddev delete --omit-snapshot
orb delete -fa
ddev restart
time ddev import-db --file=<file>

**Comment by rfay** [comment]
It would really be worthwhile to create a dummy large db dump and a standardized test for this. How about creating a repository specifically for this class of testing?

**Comment by douggreen** [comment]
Pretty much no change with the 100MB log file

* Before 112.55s user 12.05s system 1% cpu 2:44:06.85 total
* After 108.91s user 10.44s system 1% cpu 1:59:04.68 total * I think this was a bad test *, I did another run of this which showed 2:15-ish
* 100MB log file 109.22s user 11.21s system 1% cpu 2:18:28.08 total

**Comment by hkirsman** [comment]
And there are no extra hooks or other settings anymore with the above tests? I did create totally new environment for myself to be sure and as suggested above (wondering if also something else could affect the db settings eg type drupal 10):

```
name: example.com
type: drupal10
docroot: web
php_version: "8.2"
webserver_type: nginx-fpm
xdebug_enabled: false
additional_hostnames: []
additional_fqdns: []
database:
    type: mysql
    version: "8.0"
use_dns_when_possible: true
composer_version: "2"
web_environment: []
corepack_enable: false

```

Also, my OS is Linux, running on ThinkPad X1 gen 9.

[rfay](https://github.com/rfay) I do agree on the dummy db.

**Comment by douggreen** [comment]
innodb_write_io_threads=32
*  109.10s user 11.20s system 1% cpu 2:11:09.49 total
so threads makes almost no difference, remove innodb_write_io_threads, so using the default innodb_write_io_threads=4
* 109.86s user 11.40s system 1% cpu 2:16:13.53 total

**Comment by seebeen** [comment]
@rfay I can optimize the default MySQL 8.0 config DDEV ships with so it provides better performance for general use cases (php apps - primarily drupal and WP)

**Comment by rfay** [comment]
Clear improvements are welcome. And the folks in this thread can test it. Mostly complaints are usually about import speed.

**Comment by rfay** [comment]
The only thing I've heard that's possibly conclusive here is that @hkirsman believe this is a significant improvement:

```
innodb_write_io_threads=16
innodb_doublewrite=0
``

Is that true? Any followup? I was hoping for a PR.

**Comment by seebeen** [comment]
Here is the config which provides an immense boost.

```
#
# * Keys and buffers
#
key_buffer_size         = 0
sort_buffer_size        = 2M
join_buffer_size        = 128M

#
# * InnoDB
#
innodb_fast_shutdown            = 1
innodb_max_dirty_pages_pct      = 90
innodb_max_dirty_pages_pct_lwm  = 10
innodb_doublewrite              = 0
innodb_log_compressed_pages     = 0
innodb_lru_scan_depth           = 1024
innodb_page_cleaners            = 8
innodb_flush_neighbors          = 0
innodb_adaptive_flushing        = 1
innodb_flush_log_at_trx_commit  = 2
innodb_write_io_threads         = 8
innodb_read_io_threads          = 8
innodb_flush_method             = O_DIRECT_NO_FSYNC
innodb_use_native_aio           = 1
innodb_io_capacity              = 40000
innodb_io_capacity_max          = 50000
innodb_compression_level        = 2
innodb_stats_on_metadata        = 0
```

**Comment by douggreen** [comment]
tl;dr; the only proposed change that had an impact on importing (ddev
import-db) was setting innodb_doublewrite=0 or setting
innodb_write_io_threads to match the number of cores. The latest
additional changes in this thread made importing slower.

Some quick testing on one of our smaller .sql files.

# Standardizing the tests:

1. ddev delete --omit-snapshot -a && orb delete -fa
2. Then using CleanMyMacX to "Free Up Purgeable Space" and "Run
Maintenance Scripts"
3. Modify .ddev/mysql/myproject.cnf
4. orb start && ddev restart

I've removed our post processing hooks (drush updb -y && drush en -y
dblog stage_file_proxy && drush cc all).

hooks:
  pre-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE DISABLE INNODB REDO_LOG;"
  post-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE ENABLE INNODB REDO_LOG;"

# The machine

Apple M1 Pro, 32GB RAM, 10 cores (8 performance and 2 efficiency),
Sonoma 14.5, ddev v1.23.3, about 500GB free disk space

# Baseline Config

```
[mysqld]
max_connections=200
tmp_table_size=64M
max_heap_table_size=64M
sql-mode=NO_ENGINE_SUBSTITUTION
```

# The tests

These tests were run with ```time ddev import-db
--file=dbsuperdiet.sql.gz```. This is a Drupal database dump that is
1.9G compressed. Time's below are total time only:

Test 0 - 8:43.18 - Baseline
Test 1 - 8:46.28 - change pre/post to not stop INNODB logging
Test 2 - 8:52.79 - add innodb_write_io_threads=16 only
Test 3 - 8:16.05 - add innodb_write_io_threads=8 only
Test 3 - 7:39.62 - add innodb_write_io_threads=10 only -
Test 5 - 7:48.83 - add innodb_doublewrite=0 only - this is surprising also,
Test 6 - 8:12.34 - add innodb_write_io_threads=16 &
innodb_doublewrite=0 - this has become our default since this thread
was first reported
Test 7 - 7:37.81 - add innodb_write_io_threads=10 &
innodb_doublewrite=0 - not much better than 3 or 5
Test 8 - 8:34.94 - add all the changes from below
Test 9 - 7:50.18 - changes from below but matching my number of cores of 10

# Conclusion

1. My prior strategy of disabling and re-enabling INNODB had little impact.
2. The additional proposed changes make it slower.
3. If you do set any of the thread counts, don't exceed the number of cores.
4. Setting innodb_doublewrite=0 by itself is almost helpful as
anything and doesn't rely on the number of cores, so this is something
we can do in git shared configs.

I'll test the change to just remove setting innodb_write_io threads
with our larger database this weekend (44G, Test 3 above, The baseline
of 2:21:40.86 run just a few days ago is Test 6 above).

Doug

On Jul 19, 2024, at 3:48 AM, Sibin Grašić ***@***.***> wrote:

﻿

Here is the config which provides an immense boost.

#
# * Keys and buffers
#
key_buffer_size         = 0
sort_buffer_size        = 2M
join_buffer_size        = 128M

#
# * InnoDB
#
innodb_fast_shutdown            = 1
innodb_max_dirty_pages_pct      = 90
innodb_max_dirty_pages_pct_lwm  = 10
innodb_doublewrite              = 0
innodb_log_compressed_pages     = 0
innodb_lru_scan_depth           = 1024
innodb_page_cleaners            = 8
innodb_flush_neighbors          = 0
innodb_adaptive_flushing        = 1
innodb_flush_log_at_trx_commit  = 2
innodb_write_io_threads         = 8
innodb_read_io_threads          = 8
innodb_flush_method             = O_DIRECT_NO_FSYNC
innodb_use_native_aio           = 1
innodb_io_capacity              = 40000
innodb_io_capacity_max          = 50000
innodb_compression_level        = 2
innodb_stats_on_metadata        = 0

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you were mentioned.Message ID:
***@***.***>

**Comment by rfay** [comment]
It looks to me like `innodb_doublewrite=0` is a safe and reasonable thing for a local development environment:

* https://mariadb.com/kb/en/innodb-doublewrite-buffer/
* https://dba.stackexchange.com/questions/86636/when-is-it-safe-to-disable-innodb-doublewrite-buffering

**Comment by rfay** [comment]
See the PR, please do test it and provide your results, artifacts and instructions at https://github.com/ddev/ddev/pull/6411#issuecomment-2239288221

Here are my results: https://docs.google.com/spreadsheets/d/1a1eZVexjQZU4jbnwQxSir01X4n7K6M0jtDkT5T8lib0/edit?usp=sharing

It gives a modest improvement in MySQL 8.0 and MySQL 5.7, about 20% improvement - 437s instead of 551s

The improvement may be different based on the type of import, especially its structure; this is a random `hugedb` that I keep around for this kind of testing, and may not have the same structure as your databases.

**Comment by douggreen** [comment]
On the MariaDB 10.11 line, is this showing the innodb_doublewrite=0 has no
impact (actually a few seconds slower)?

**Comment by rfay** [comment]
> On the MariaDB 10.11 line, is this showing the innodb_doublewrite=0 has no

Agreed. The difference in timing is certainly not statistically significant. (Each of these was tried only one time; in general it didn't seem to affect mariadb)

**Comment by douggreen** [comment]
We are using mysql 8.0. It seems like we should switch to mariadb 10.11, or
at least test it. Do you agree?

**Comment by rfay** [comment]
> We are using mysql 8.0. It seems like we should switch to mariadb 10.11, or at least test it. Do you agree?

As far as what your project chooses (I think that's what you're asking) it is probably a choice based on the capabilities of MySQL 8.0. These databases have diverged a bit over the years, with different features and strengths (and weaknesses). 

For development import time can be important, but this is unusual for production, and production requirements usually win.

**Comment by douggreen** [comment]
I didn't test the PR. But I think this change is solid.

I wondered if any single test can really tell us anything. So I wrote
a script to run the import 10 times. The script restarts ddev between
each run.

So I ran a baseline test of 10 runs on the smaller database:

Here is the baseline,

* results: 8:35.53, 8:38.97, 9:04.61, 8:16.03, 7:44.10, 7:52.91,
7:33.34, 7:42.74, 8:12.99, 8:30.87
* average: 8:13.21
* mean: 8:15.51
* stddev: 28.11 seconds

And then deleted dde & orb, free'd up purgeable memory, and then
re-ran the test with innodb_doublewrite=0

* results: 7:40.57, 7:17.47, 8:33.01, 7:52.60, 7:15.46, 7:08.89,
7:43.72, 8:01.68, 7:18.94, 8:04.33
* average: 7:41.67
* mean: 7:42.14
* stddev: 25.57

I want to do a similar test on our larger database (6.8G compressed,
44G uncompressed), but that takes 2 hours+ per run. I did do a single
import with and without innodb_doublewrite=0. Before was 2:20:28.21
and after was 2:01:14.75.

Doug

**Comment by rfay** [comment]
@hkirsman @douggreen @seebeen I pulled
* https://github.com/ddev/ddev/pull/6411

Would love to have you test [HEAD](https://ddev.readthedocs.io/en/stable/developers/building-contributing/#testing-latest-commits-on-head) now, it's super easy. 

If you discover specific improvements in the future please open a new issue with the specific suggestion. It seemed that `innodb_doublewrite=0` was the only one discovered in this round. It definitely provided improvement on import, but more like 20% for me rather than the hoped-for 40%.

**Comment by rfay** [comment]
Followups on this would include 
* A standardized large mariadb/mysql database (or perhaps more than one size) that has no personal information in it, but does have diverse structure.
* A script that can run automated timing on a database like that, with different configurations.

**Comment by rfay** [comment]
This issue was explicitly about mysql:8.0, as many people have to use that. Yes, the PR is already in DDEV v1.23.4.

**Comment by rfay** [comment]
Another discussion in 
* https://github.com/orgs/ddev/discussions/6591

**Comment by vermario** [comment]
Hello, I post here after an initial discussion on drupal ddev slack (https://drupal.slack.com/archives/C5TQRQZRR/p1731656603549439) 

We have a project with a large database (the dump is 15G uncompressed). When importing the db dump into ddev using ddev/ddev-dbserver-mysql-8.0:v1.23.5 , I noticed that the resulting volume in docker was much bigger than that, expecially because I had to do the import multiple times (to test something unrelated).

The db is 15 gigabytes, but the volume ended up taking 85 gigabytes, because of quite many big binlog. files: 

![Image](https://github.com/user-attachments/assets/f79a35f0-25bd-4b7f-b09f-b7b83cc6ce0f)

This was most likely caused by my tests of importing the database then deleting it, and there is this setting currently that would delete the binlog after 1 day: https://github.com/ddev/ddev/blob/master/containers/ddev-dbserver/files/etc/my.cnf#L31

Now, I can disable this in my particular project by creating a small configuration file in `.ddev/mysql/myconf.cnf` (For example) and adding this: 
```
[mysqld]
disable-log-bin=1
```

and after that:
1. my import operation got faster (6m 30s before, 5m 04s afterwards)
2. the disk space stayed at 20gigs

So my question is: are there disadvantages in disabling the binary log in mysql? Reading about it on the web, it seems that it's used for multi instance databases, recovery etc.. things that in local development don't sound very important.

Also in my test ddev snapshot seems to work happily without it.

So maybe we can save some CPU cycles and disk space by disabling it altogether?

Or maybe we can set a lower time for the binlog to be cleaned up with the newer `binlog_expire_logs_seconds` setting advertised here? https://dev.mysql.com/worklog/task/?id=10924

**Comment by stasadev** [comment]
I see that binlog setting was used here a long time ago:

- #1515

> So my question is: are there disadvantages in disabling the binary log in mysql? Reading about it on the web, it seems that it's used for multi instance databases, recovery etc.. things that in local development don't sound very important.

Agreed, or if these logs are needed, they should be removed when doing database import (use `RESET MASTER;` ?):

- https://dba.stackexchange.com/a/47048

**Comment by rfay** [comment]
I think the binlogs are mostly intended for replication and crash recovery. I'm pretty sure we could turn them off or limit them more. A PR is welcome. Go ahead and test with your own system, and when happy, suggest an improved approach.

**Comment by douggreen** [comment]
I use this in my config.yaml, which probably does the same thing.

hooks:
  pre-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE DISABLE INNODB REDO_LOG;"
  post-import-db:
  - exec: mysql -uroot -proot -e "ALTER INSTANCE ENABLE INNODB REDO_LOG;"

**Comment by douggreen** [comment]
... which does the same thing, but only during the import.

**Comment by stasadev** [comment]
> I use this in my `config.yaml`, which probably does the same thing. 

I tried to format the reply into a readable format, but without success (`Sent via email` doesn't allow formatting?):

Copying it here:

```
hooks:
    pre-import-db:
        - exec: mysql -uroot -proot -e "ALTER INSTANCE DISABLE INNODB REDO_LOG;"
    post-import-db:
        - exec: mysql -uroot -proot -e "ALTER INSTANCE ENABLE INNODB REDO_LOG;"
```

**Comment by rfay** [comment]
Update with about a 30% improvement for MySQL 8.0 imports, and a testing framework:

https://ddev.com/blog/database-improvements/
